libtransact/src/execution/executor/internal.rs

```diff
@@ -344,8 +353,26 @@ impl ExecutorThread {
                                 });
                         }
 
+                        Ok(ExecutorCommand::CreateReader(task_iterator, notifier)) => {
+                            let index = readers.keys().max().cloned().unwrap_or(0);
```

Comment:
Shouldn't this be max + 1?

Reply:
It should, though the original code had the same issue - will fix in an additional commit

Reply:
I had assumed that was on purpose to reset the indexes if they had all been removed.